tests/stack/test_api.py

```diff
@@ -338,3 +338,65 @@ def test_deploy_resource(self, stack_api):
         }
         with pytest.raises(StackError):
             stack_api._deploy_resource(resource_badtype)
+
+    def test_download_workspace(self, stack_api, tmpdir):
```

Comment:
Another set of important behaviors to test are how "overwrite" is handled in different scenarios

Reply:
Feel like this is something that should be handled in the `workspace` API. I planned to just check in this test that the right arguments are plugged into the `workspace` API `import_workspace` and `import_workspace_dir` functions.